test/usecases/recursion/recursion.mod

```diff
@@ -0,0 +1,34 @@
+NEURON {
+    SUFFIX recursion
+}
+
+: single recursive call
+FUNCTION f_simple(n) {
+    f_simple = f_simple(n)
+}
+
+: multiple recursive calls
+FUNCTION fc(n) {
+    fc = fc(fc(fc(fc(n))))
```

Comment:
This looks like infinite recursion. Why do you want to assert that this is valid?

Reply:
As mentioned, this is here to make sure that it actually compiles, there are no run-time guarantees (hence it's not present in `simulate.py`).

Reply:
Why do we want it to compile? I'd argue that we should be allowed to at least make an attempt to reject such MOD files (yes, yes halting-problem bla bla). This leaves a trail suggesting that this is essential functionality. Which is plausible, because there might be some MOD file that has an `fc` and like here never calls it.

Second issue I have: If you can't test the behaviour of the code how do you know that whatever bug you're trying to squash got squashed? (NMODL could simply replace it with a `""` and things would compile.)